### ISSUE TITLE
As a user I can add a new room, whare and whanau and edit existing records

### ISSUE THREAD

**Issue by MeghanHughes** [WhareHauora/App#40]
Set up Set Up pages to CUD rooms, whare, whanau
[The files are located in >components>profile>setUp](https://github.com/WhareHauora/App/tree/Development/components/profile/setUp)

**Comment by MeghanHughes** [comment]
Need to fix the button styling - had to change <a> to <button> to get onClick happening
![screenshot at 2018-04-20 12-00-57](https://user-images.githubusercontent.com/22287063/39024054-8222aa0e-4492-11e8-8859-a3a5bb06cc5b.png)

**Comment by MeghanHughes** [comment]
@Br3nda has added an api method for posting a new room so let's get that one up.
https://documenter.getpostman.com/view/3560032/wharehauora/RVnb9wYN#0f1885d2-22d1-172a-fbb2-90856ba733a7

**Comment by MeghanHughes** [comment]
So - we use [Final Form] (https://github.com/final-form/react-final-form#examples) as the component library for the form elements.

**Comment by MeghanHughes** [comment]
@Br3nda could you document these CRUD routes in the Postman docs?

**Comment by janie-t** [comment]
Currently no component for Adding a new Whare in the Rooms & Whare page.

**Comment by MeghanHughes** [comment]
Yeah there is @janie-t it is in `Components/profile/set-up/AddWhareModal` 
it is activated by a ternary in `WhareList` which is imported in to `Rooms` but I have just noticed that when it expands the page doesn't move down so it looks like it doesn't open anything...

**Comment by MeghanHughes** [comment]
Also need to hid the Add Whare/Room buttons when the form renders... I'll have a go at that today

**Comment by Br3nda** [comment]
Sorry i've been without any time to update the docs. It does follow the jsonapi.org spec.

**Comment by Br3nda** [comment]
The api PR is merged and on production. The postman docs have examples now.

**Comment by MeghanHughes** [comment]
Hey @Br3nda do I have rights to add to the database? I am just testing the CRUD methods and I am getting a 401 trying to POST to the whare route
eg
```
handleClick(event) {
    axios({
      method: "post",
      url: `https://app.wharehauora.nz/api/v1/homes?access_token=${Cookies.get('token')}`,
      data: {
        data: {
          type: "homes",
          attributes: {
        		"name": this.state.name
        	}
        }
      },
      headers: {'Content-Type': 'application/vnd.api+json', 'Accept': 'application/vnd.api+json'}
    })
  }
```

**Comment by Br3nda** [comment]
Most home CRUD methods require you to be the owner or janitor
I've been testing as a janitor -- but looks like users can't create their own homes.
I can have a look at the policies. If you wanted to change them yourself that's cool too - it's this file:
https://github.com/WhareHauora/wharehauora-server/blob/master/app/policies/home_policy.rb

**Comment by janie-t** [comment]
@MeghanHughes are you going to change that home_policy.rb doc?

**Comment by MeghanHughes** [comment]
I'm going to look into it now @janie-t

**Comment by MeghanHughes** [comment]
ahhhhhhhhh no lolol - had a look but don't want to touch it. @Br3nda let me know when you've had a chance to give users access and I will just make sure the methods are set up my end.
